src/expr/src/linear.rs

```diff
@@ -1161,6 +1161,27 @@ pub mod plan {
         /// As the evaluation exits early with failed predicates, it may
         /// miss some errors that would occur later in evaluation.
         #[inline(always)]
+        pub fn evaluate_into<'a>(
```

Comment:
I'd add some doc comments here about `row` and what happens. E.g., in this case "not cleared first, but emptied if the result is `Ok(Some(_))`". I'd be tempted to tweak the logic to try and make that more consistent (perhaps empty it first, so that we never end up with half-filled data?), or very explicitly document it as containing "undefined-ish" contents after the call.

Reply:
I added a comment to describe this behavior, and added a call to `Row::clean()` before extending.